### PR TITLE
Added null check to CorsPolicyBuilder

### PR DESCRIPTION
**Pull request by chrispickford**
It was noticed that there is no null checking in the `CorsPolicyBuilder.GetNormalizedOrigin` method after a `NullReferenceException` was picked up. In the case that a null string is passed to the method via `WithOrigins` (in our case it was because a configuration file read failed) this change throws an `ArgumentNullException` which gives a clearer indication of what has failed.

Addresses #19830

### CODE REVIEW

**Review by analogrelay — APPROVED** [review]
Seems reasonable to me. Thanks for the fix!

**Review by analogrelay — DISMISSED** [review]
Looks like the test you added is [failing](https://dev.azure.com/dnceng/public/_build/results?buildId=558491&view=ms.vss-test-web.build-test-results-tab&runId=17610246&resultId=103827&paneView=debug):

```
Assert.Throws() Failure

Expected: typeof(System.ArgumentNullException)

Actual: typeof(System.NullReferenceException): Object reference not set to an instance of an object.

---- System.NullReferenceException : Object reference not set to an instance of an object.
```

Looking at [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+WRilQIIAUAlLfY/wFkOAOwCuAG3FcA3HwYBfOaSUlyrIQAcAhlC0BbAM7kcABgDaAXVIBLAC4xDPOgX6KC8oA=), what your test actually does is pass a *null array* rather than an array with a null item.

Working with `params` is a little tricky, as you can see since I approved the change without realizing this would happen! Good thing we have tests!

I'd suggest a few changes here:

1. Add a null-check directly on `WithOrigins` and keep the current test you've added

1. Add a second test that explicitly constructs an array with an empty element and passes it to `WithOrigins` (i.e. `.WithOrigins(new string[] { "foo", null, "bar" })`) and validate that the inner `ArgumentNullException` is properly thrown.

I think that will resolve the issue and add a little more coverage here.

**Review by mkArtakMSFT — APPROVED** [review]

**Review by pranavkm — APPROVED** [review]

**Comment by mkArtakMSFT** [comment]
@chrispickford you still haven't addressed @anurse 's comment regarding what you're actually testing. Your test doesn't match your change. Do you want to give this another try?

**Comment by chrispickford** [comment]
@mkArtakMSFT @anurse Apologies for the delay, I've updated the test and added extra null check.

**Comment by chrispickford** [comment]
@anurse @Tratcher I'm not sure what's going on with the CI build. Any pointers?

**Comment by mkArtakMSFT** [comment]
Closed and reopened the PR to re-trigger the build

**Comment by mkArtakMSFT** [comment]
@anurse this is blocked on you to sign-off.

**Comment by anurse** [comment]
I’m not able to do it from my phone but I’m happy to have my review dismissed to unblock merging if there are other approving reviews.

**Pull request merged**
